src/Npgsql/NpgsqlTypes/NpgsqlDate.cs

```diff
@@ -1,452 +1,235 @@
-                InternalType.NegativeInfinity => other._type == InternalType.NegativeInfinity,
-                InternalType.Finite           => other._type == InternalType.Finite && _daysSinceEra == other._daysSinceEra,
-                _ => false
-            };
-
-        public override bool Equals(object? obj) => obj is NpgsqlDate date && Equals(date);
+        /// <summary>
+        /// The value of this constant is equivalent to January 1, 2000, in the Gregorian calendar.
+        /// <see cref="PostgreSqlEpoch"/> defines the point in time when PostgreSQL time is equal to 0.
+        /// </summary>
+        public static NpgsqlDate PostgreSqlEpoch => new NpgsqlDate(0);
 
-        public int CompareTo(NpgsqlDate other)
-            => _type switch
-            {
-                InternalType.Infinity         => other._type == InternalType.Infinity ? 0 : 1,
-                InternalType.NegativeInfinity => other._type == InternalType.NegativeInfinity ? 0 : -1,
-                _ => other._type switch
-                {
-                    InternalType.Infinity         => -1,
-                    InternalType.NegativeInfinity => 1,
-                    _                             => _daysSinceEra.CompareTo(other._daysSinceEra)
-                }
-            };
+        /// <summary>
+        /// The value of this constant is equivalent to January 1, 1970, in the Gregorian calendar.
+        /// <see cref="UnixEpoch"/> defines the point in time when Unix time is equal to 0.
+        /// </summary>
+        public static NpgsqlDate UnixEpoch => new NpgsqlDate(-10957);
 
-        public int CompareTo(object? o)
-            => o == null
-                ? 1
-                : o is NpgsqlDate npgsqlDate
-                    ? CompareTo(npgsqlDate)
-                    : throw new ArgumentException();
+        /// <summary>Gets the year component of the date represented by this instance.</summary>
+        /// <value>The year component.</value>
+        public int Year => NpgsqlDateTime.GetDatePart(Days, NpgsqlDateTimePart.Year);
 
-        public override int GetHashCode() => _daysSinceEra;
+        /// <summary>Gets the month component of the date represented by this instance.</summary>
+        /// <value>The month component, expressed as a value between 1 and 12.</value>
+        public int Month => NpgsqlDateTime.GetDatePart(Days, NpgsqlDateTimePart.Month);
 
-        #endregion
+        /// <summary>Gets the day component of the date represented by this instance.</summary>
+        /// <value>The day component, expressed as a value between 1 and 31.</value>
+        public int Day => NpgsqlDateTime.GetDatePart(Days, NpgsqlDateTimePart.Day);
 
-        #region Operators
+        /// <summary>Gets the number of days since PostgreSQL epoch.</summary>
+        /// <value>The number of days since PostgreSQL epoch.</value>
+        /// <seealso cref="PostgreSqlEpoch"/>
+        public int Days => _days;
```

Comment:
As elsewhere, I don't think we should expose this.

Reply:
Will make it internal, but won't make it an auto property. When it get's merged and hope ranges too, I will continue with type handlers again. For them I made an option to get a buffer as a span, so making `TryWrite`/`TryRead` methods would completely remove all logic into types. So handlers would work like "here, take your N byte span sent by the backend and do your dirty things I know nothing about".

Reply:
OK. I really only care about the public surface exposed by these types - whatever internal mechanisms are used between them and their handlers is less important.